httplib.h

```diff
@@ -5064,7 +5078,17 @@ Server::process_request(Stream &strm, bool close_connection,
   }
 
   // Rounting
-  if (routing(req, res, strm)) {
+  bool routed;
+  try {
+    routed = routing(req, res, strm);
+  } catch (std::exception & e) {
+    if (exception_handler_)
+      routed = exception_handler_(req, res, e) == HandlerResponse::Handled;
+    else
+      throw e;
```

Comment:
`e` should never be re-thrown here. The original code (Ln 4848-4853) should be moved here instead. If a user really wants to re-throw it (it's dangerous though...), it should be re-thrown from a user's exception handler.